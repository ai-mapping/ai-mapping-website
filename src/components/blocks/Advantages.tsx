import React from 'react';
import FastDeliveryIcon from '@/components/ui/icons/FastDeliveryIcon';
import QualityIcon from '@/components/ui/icons/QualityIcon';
import SearchIcon from '@/components/ui/icons/SearchIcon';
import PriceIcon from '@/components/ui/icons/PriceIcon';
import CheckmarkIcon from '@/components/ui/icons/CheckmarkIcon';
import Section from '@/components/layout/Section';

type AdvantageItem = {
  icon: React.ReactNode;
  title: string;
  description: string;
  features: string[];
};

export type AdvantagesProps = {
  title: string;
  subtitle: string;
  advantages: AdvantageItem[];
};

// Default values
export const defaultAdvantagesProps: AdvantagesProps = {
  title: 'Наши преимущества',
  subtitle: 'Почему клиенты выбирают нас',
  advantages: [
    {
      icon: <FastDeliveryIcon />,
      title: 'Экспресс-доставка в условиях санкций',
      description: 'В то время как официальные дилеры ушли с рынка, мы создали эффективные каналы доставки запчастей из Китая в короткие сроки.',
      features: [
        '7-14 дней на доставку вместо 1-3 месяцев',
        'Собственная сеть представителей в Китае',
        'Прямые контакты с производителями'
      ]
    },
    {
      icon: <QualityIcon />,
      title: 'Высокое качество неоригинальных запчастей',
      description: 'Мы поставляем только качественные аналоги, которые сравнимы с оригинальными запчастями по надежности и долговечности.',
      features: [
        'Строгий контроль качества на производстве',
        'Соответствие техническим спецификациям',
        'Экономия до 30-40% от стоимости оригинала'
      ]
    },
    {
      icon: <SearchIcon />,
      title: 'Поиск и доставка редких запчастей',
      description: 'Мы специализируемся на поиске редких и труднодоступных деталей, которые невозможно найти у других поставщиков.',
      features: [
        'Доступ к широкой сети производителей в Китае',
        'Возможность изготовления под заказ',
        'Решение нестандартных задач по поиску запчастей'
      ]
    },
    {
      icon: <PriceIcon />,
      title: 'Ценовое преимущество',
      description: 'Мы предлагаем конкурентные рыночные цены и оптимальное соотношение цена/качество для представленных запчастей.',
      features: [
        'Экономия от 20% до 40% по сравнению с оригиналом',
        'Скидки для постоянных клиентов и при крупных заказах',
        'Прозрачное ценообразование'
      ]
    }
  ]
};

const Advantages = ({
  title = defaultAdvantagesProps.title,
  subtitle = defaultAdvantagesProps.subtitle,
  advantages = defaultAdvantagesProps.advantages
}: Partial<AdvantagesProps>) => {
  // Split advantages into two rows
  const firstRow = advantages.slice(0, 2);
  const secondRow = advantages.slice(2, 4);

  return (
    <Section background="background" paddingY="lg" containerWidth="container">
        <h2 className="text-2xl md:text-3xl font-semibold text-center mb-2">
          {title}
        </h2>
        <p className="text-dark text-center mb-8">
          {subtitle}
        </p>

        <div className="grid grid-cols-1 md:grid-cols-2 gap-8">
          {firstRow.map((advantage, index) => (
            <div key={`advantage-1-${index}`} className="bg-white rounded-lg shadow-md p-6">
              <div className="flex items-start">
                <div className="text-primary mr-4 mt-1">
                  {advantage.icon}
                </div>
                <div>
                  <h3 className="text-xl font-semibold mb-3">
                    {advantage.title}
                  </h3>
                  <p className="text-dark mb-4">
                    {advantage.description}
                  </p>
                  <ul className="text-dark space-y-2">
                    {advantage.features.map((feature, featureIndex) => (
                      <li key={`feature-1-${index}-${featureIndex}`} className="flex items-start">
                        <CheckmarkIcon />
                        <span>{feature}</span>
                      </li>
                    ))}
                  </ul>
                </div>
              </div>
            </div>
          ))}
        </div>

        <div className="mt-8 grid grid-cols-1 md:grid-cols-2 gap-8">
          {secondRow.map((advantage, index) => (
            <div key={`advantage-2-${index}`} className="bg-white rounded-lg shadow-md p-6">
              <div className="flex items-start">
                <div className="text-primary mr-4 mt-1">
                  {advantage.icon}
                </div>
                <div>
                  <h3 className="text-xl font-semibold mb-3">
                    {advantage.title}
                  </h3>
                  <p className="text-dark mb-4">
                    {advantage.description}
                  </p>
                  <ul className="text-dark space-y-2">
                    {advantage.features.map((feature, featureIndex) => (
                      <li key={`feature-2-${index}-${featureIndex}`} className="flex items-start">
                        <CheckmarkIcon />
                        <span>{feature}</span>
                      </li>
                    ))}
                  </ul>
                </div>
              </div>
            </div>
          ))}
        </div>
    </Section>
  );
};

export default Advantages;
