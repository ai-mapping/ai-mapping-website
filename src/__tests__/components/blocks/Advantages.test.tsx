import React from 'react';
import { render, screen } from '@testing-library/react';
import Advantages, { defaultAdvantagesProps } from '@/components/blocks/Advantages';

describe('Advantages Component', () => {
  it('renders with default props correctly', () => {
    const { container } = render(<Advantages />);
    
    // Check if title and subtitle are rendered
    expect(screen.getByText(defaultAdvantagesProps.title)).toBeInTheDocument();
    expect(screen.getByText(defaultAdvantagesProps.subtitle)).toBeInTheDocument();
    
    // Check if all advantages are rendered
    defaultAdvantagesProps.advantages.forEach((advantage) => {
      expect(screen.getByText(advantage.title)).toBeInTheDocument();
      expect(screen.getByText(advantage.description)).toBeInTheDocument();
      
      // Check if all features are rendered
      advantage.features.forEach((feature) => {
        expect(screen.getByText(feature)).toBeInTheDocument();
      });
    });
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom props correctly', () => {
    const customProps = {
      title: 'Custom Title',
      subtitle: 'Custom Subtitle',
      advantages: [
        {
          icon: <div data-testid="custom-icon" />,
          title: 'Custom Advantage',
          description: 'Custom description text',
          features: ['Feature 1', 'Feature 2']
        }
      ]
    };
    
    const { container } = render(<Advantages {...customProps} />);
    
    expect(screen.getByText(customProps.title)).toBeInTheDocument();
    expect(screen.getByText(customProps.subtitle)).toBeInTheDocument();
    expect(screen.getByText(customProps.advantages[0].title)).toBeInTheDocument();
    expect(screen.getByText(customProps.advantages[0].description)).toBeInTheDocument();
    customProps.advantages[0].features.forEach(feature => {
      expect(screen.getByText(feature)).toBeInTheDocument();
    });
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});