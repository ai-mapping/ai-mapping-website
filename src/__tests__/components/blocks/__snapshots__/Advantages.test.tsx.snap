// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advantages Component renders with custom props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8"
    >
      <h2
        class="text-2xl md:text-3xl font-semibold text-center mb-2"
      >
        Custom Title
      </h2>
      <p
        class="text-dark text-center mb-8"
      >
        Custom Subtitle
      </p>
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-8"
      >
        <div
          class="bg-white rounded-lg shadow-md p-6"
        >
          <div
            class="flex items-start"
          >
            <div
              class="text-primary mr-4 mt-1"
            >
              <div
                data-testid="custom-icon"
              />
            </div>
            <div>
              <h3
                class="text-xl font-semibold mb-3"
              >
                Custom Advantage
              </h3>
              <p
                class="text-dark mb-4"
              >
                Custom description text
              </p>
              <ul
                class="text-dark space-y-2"
              >
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Feature 1
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Feature 2
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-8 grid grid-cols-1 md:grid-cols-2 gap-8"
      />
    </div>
  </section>
</div>
`;

exports[`Advantages Component renders with default props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8"
    >
      <h2
        class="text-2xl md:text-3xl font-semibold text-center mb-2"
      >
        Наши преимущества
      </h2>
      <p
        class="text-dark text-center mb-8"
      >
        Почему клиенты выбирают нас
      </p>
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-8"
      >
        <div
          class="bg-white rounded-lg shadow-md p-6"
        >
          <div
            class="flex items-start"
          >
            <div
              class="text-primary mr-4 mt-1"
            >
              <div
                class="h-10 w-10"
                data-testid="FastDelivery-icon"
              >
                FastDelivery
              </div>
            </div>
            <div>
              <h3
                class="text-xl font-semibold mb-3"
              >
                Экспресс-доставка в условиях санкций
              </h3>
              <p
                class="text-dark mb-4"
              >
                В то время как официальные дилеры ушли с рынка, мы создали эффективные каналы доставки запчастей из Китая в короткие сроки.
              </p>
              <ul
                class="text-dark space-y-2"
              >
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    7-14 дней на доставку вместо 1-3 месяцев
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Собственная сеть представителей в Китае
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Прямые контакты с производителями
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="bg-white rounded-lg shadow-md p-6"
        >
          <div
            class="flex items-start"
          >
            <div
              class="text-primary mr-4 mt-1"
            >
              <div
                class="h-10 w-10"
                data-testid="Quality-icon"
              >
                Quality
              </div>
            </div>
            <div>
              <h3
                class="text-xl font-semibold mb-3"
              >
                Высокое качество неоригинальных запчастей
              </h3>
              <p
                class="text-dark mb-4"
              >
                Мы поставляем только качественные аналоги, которые сравнимы с оригинальными запчастями по надежности и долговечности.
              </p>
              <ul
                class="text-dark space-y-2"
              >
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Строгий контроль качества на производстве
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Соответствие техническим спецификациям
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Экономия до 30-40% от стоимости оригинала
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-8 grid grid-cols-1 md:grid-cols-2 gap-8"
      >
        <div
          class="bg-white rounded-lg shadow-md p-6"
        >
          <div
            class="flex items-start"
          >
            <div
              class="text-primary mr-4 mt-1"
            >
              <div
                class="h-10 w-10"
                data-testid="Search-icon"
              >
                Search
              </div>
            </div>
            <div>
              <h3
                class="text-xl font-semibold mb-3"
              >
                Поиск и доставка редких запчастей
              </h3>
              <p
                class="text-dark mb-4"
              >
                Мы специализируемся на поиске редких и труднодоступных деталей, которые невозможно найти у других поставщиков.
              </p>
              <ul
                class="text-dark space-y-2"
              >
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Доступ к широкой сети производителей в Китае
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Возможность изготовления под заказ
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Решение нестандартных задач по поиску запчастей
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="bg-white rounded-lg shadow-md p-6"
        >
          <div
            class="flex items-start"
          >
            <div
              class="text-primary mr-4 mt-1"
            >
              <div
                class="h-10 w-10"
                data-testid="Price-icon"
              >
                Price
              </div>
            </div>
            <div>
              <h3
                class="text-xl font-semibold mb-3"
              >
                Ценовое преимущество
              </h3>
              <p
                class="text-dark mb-4"
              >
                Мы предлагаем конкурентные рыночные цены и оптимальное соотношение цена/качество для представленных запчастей.
              </p>
              <ul
                class="text-dark space-y-2"
              >
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Экономия от 20% до 40% по сравнению с оригиналом
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Скидки для постоянных клиентов и при крупных заказах
                  </span>
                </li>
                <li
                  class="flex items-start"
                >
                  <div
                    class="h-10 w-10"
                    data-testid="Checkmark-icon"
                  >
                    Checkmark
                  </div>
                  <span>
                    Прозрачное ценообразование
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
